### PR TITLE
Ctcp 1012 part2

### PR DESCRIPTION
**Pull request by ardwalker**

### CODE REVIEW

File: app/v2/connectors/DepartureConnector.scala
Comment (drnaylor-oc) [inline app/v2/connectors/DepartureConnector.scala:65]: Remove commented code that we're not using.

**Review by drnaylor-oc — COMMENTED** [review]
The logic looks okay, but the controller shouldn't house all the logic, that caused issues in debugging and maintainability for phase 4 and we should avoid that in phase 5.

File: app/v2/controllers/stream/StreamingParsers.scala
Comment (derekspalmer) [inline app/v2/controllers/stream/StreamingParsers.scala:46]: Commented out stuff for a reason?

File: test/v2/data/TestXml.scala
Comment (derekspalmer) [inline test/v2/data/TestXml.scala:35]: ?

**Review by derekspalmer — COMMENTED** [review]
Quite a lot of code commented out - should it be removed?

File: app/v2/connectors/BaseConnector.scala
Comment (drnaylor-oc) [inline app/v2/connectors/BaseConnector.scala:38]: This string format isn't used in Phase 5, the correlation ID is simply `movementId-messageId`

File: app/models/HateaosDepartureResponse.scala
Comment (ardwalker) [reply app/models/HateaosDepartureResponse.scala:29]: Was trying to be clever by reusing the same class, but MessageType is different for v2. Have copied the whole class into v2 to make it a bit clearer.

File: app/v2/connectors/BaseConnector.scala
Comment (ardwalker) [reply app/v2/connectors/BaseConnector.scala:34]: Agree, I went through the same thought process. Chatted with @drnaylor-oc about it and decided that the differences although sometimes small could be copied into /v2 versions in their entirety, theory being that eventually v1 could be retired and most of what we use should already be there in v2. 

In this case yes, could have simply added a new variable for the new transit-movements service to retrieve departures from. Open to other ideas/suggestions, let me know.

File: app/v2/connectors/DepartureConnector.scala
Comment (ardwalker) [reply app/v2/connectors/DepartureConnector.scala]: You're right, this has been removed.

File: test/v2/data/TestXml.scala
Comment (ardwalker) [reply test/v2/data/TestXml.scala:35]: Removed

File: app/v2/connectors/BaseConnector.scala
Comment (ardwalker) [reply app/v2/connectors/BaseConnector.scala:42]: @drnaylor-oc  this format string is used for populating the header: "X-Message-Recipient".
So my assumption is you mean set the header to movementId-messageId?

Comment (drnaylor-oc) [reply app/v2/connectors/BaseConnector.scala:40]: Well, we don't actually have it at all, and we don't use the `X-Message-Recipient` header in P5. In P5, the correlation ID is part of the header, and you're right, we don't have the message ID. We just want to set it to 16 zeroes in the TS API.

File: app/v2/controllers/actions/MessageRequestAction.scala
Comment (drnaylor-oc) [inline app/v2/controllers/actions/MessageRequestAction.scala:48]: Rather than just sending the message, we should send a `Json.toJson(PresentationError.badRequest(message))` (or whatever it is)

File: app/v2/controllers/actions/MessageRequestAction.scala
Comment (drnaylor-oc) [inline app/v2/controllers/actions/MessageRequestAction.scala:42]: Same here - with some sort of message saying it's malformed.

File: app/v2/controllers/actions/MessageRequestAction.scala
Comment (drnaylor-oc) [inline app/v2/controllers/actions/MessageRequestAction.scala:45]: And here (though what do we mean by "no instructions found?").

File: app/v2/controllers/actions/ValidateDepartureMessageTypeAction.scala
Comment (drnaylor-oc) [inline app/v2/controllers/actions/ValidateDepartureMessageTypeAction.scala:34]: We should also make this fit the error structure - using the `PresentationError` again with a message stating "Support for IEXXX is not implemented"... or something like that.

File: conf/application.conf
Comment (drnaylor-oc) [inline conf/application.conf:168]: Note: we need to update this config in `app-config-base` for the in-environment values.

**Review by drnaylor-oc — COMMENTED** [review]
Mostly consistency things from me.

File: app/v2/controllers/actions/MessageRequestAction.scala
Comment (ardwalker) [reply app/v2/controllers/actions/MessageRequestAction.scala]: @drnaylor-oc  Good question, I just looked into it, seems to only apply to unloading message instructions. Are you ok if I remove all this for now, can all be added back in if/when required later on?

Comment (drnaylor-oc) [reply app/v2/controllers/actions/MessageRequestAction.scala]: Yes, that's fine - better to go simple and put it back later if we need it.

**Review by derekspalmer — APPROVED** [review]

**Comment by drnaylor-oc** [comment]
Don't merge this until the config itself is merged, otherwise we'll have to kick off a manual build of this.

(Also just testing this locally, so give me a moment to do that)

**Comment by drnaylor-oc** [comment]
I've just tried this locally and it's not working. I did the following:

* Submitted a departure declaration, got the departure ID
* Called the get departure endpoint, verified the departure was there
* Called the test support API endpoint with the following payload:

```json
{
    "message": {
        "messageType": "IE928"
    }
}
```

I got the following back:

```json
{
    "message": "Departure with ID 631f233aa410d0e2 was not found",
    "code": "NOT_FOUND"
}
```

Need to figure out why.

File: app/v2/connectors/DepartureConnector.scala
Comment (drnaylor-oc) [inline app/v2/connectors/DepartureConnector.scala:45]:
```suggestion
    val url = s"${appConfig.transitMovementsUrl}/transit-movements/traders/${eori.value}/movements/departures/${departureId.value}"
```

Comment (drnaylor-oc) [reply app/v2/connectors/DepartureConnector.scala:45]: With this, I get a 500, which is expected because the failure is that there is no endpoint on the router yet, I see it being called here: 

```
022-09-12 13:25:32,243 level=[DEBUG] logger=[connector] thread=[scala-execution-context-global-202] rid=[] user=[] message=[:POST:421225881:421.225ms:1110293001:1.110s:96d3:http://localhost:9478/traders/99999912345/movements/631f233aa410d0e2-0000000000000000/messages/:ok] 
```

Comment (liandra-giacomine) [reply app/v2/connectors/DepartureConnector.scala:45]: @drnaylor-oc transitMovementsUrl evaluates to transit-movements, so this change would add it again?

Comment (drnaylor-oc) [reply app/v2/connectors/DepartureConnector.scala:45]: It's the change I had to make on my local copy to get this working. `appConfig.transitMovementsUrl` appears to evaluate to `http://localhost:9520` only.

Comment (liandra-giacomine) [reply app/v2/connectors/DepartureConnector.scala:45]: Yeah you're right, it evaluates to $protocol://$host:$port

File: app/v2/connectors/InboundRouterConnector.scala
Comment (drnaylor-oc) [inline app/v2/connectors/InboundRouterConnector.scala:41]: I was just comparing this with the EIS -> MDTP spec, and this is slightly wrong.

```suggestion
    val url = appConfig.transitMovementsRouterUrl + s"/transit-movements-router/movements/${departureId.value}-0000000000000000/messages/"
```

The EIS->MDTP spec also looks wrong in this regard because all our microservices, barring the actual APIs, have their route prefixed with the name of the service, something the spec doesn't have in it.

File: app/v2/connectors/BaseConnector.scala
Comment (liandra-giacomine) [inline app/v2/connectors/BaseConnector.scala:37]: this can be a lazy val

File: app/v2/connectors/BaseConnector.scala
Comment (liandra-giacomine) [inline app/v2/connectors/BaseConnector.scala:44]: You don't need to set the val here

File: app/v2/connectors/InboundRouterConnector.scala
Comment (liandra-giacomine) [inline app/v2/connectors/InboundRouterConnector.scala:41]: also need to add /transit-movements/ in here

Comment (ardwalker) [reply app/v2/connectors/InboundRouterConnector.scala:41]: @liandra-giacomine , which part?

**Pull request merged**
